js/index.js

```diff
@@ -42,42 +57,119 @@ function renderPepperoni() {
   });
 }
 
+
+
+// Iteration 1: set the visibility of `<section class="mushroom">`
 function renderMushrooms() {
-  // Iteration 1: set the visibility of `<section class="mushroom">`
+  document.querySelectorAll('.mushroom').forEach(function (oneMushroom) {
+    if (state.mushrooms === true) {
+      oneMushroom.style.visibility = 'visible';
+    } else {
+      oneMushroom.style.visibility = 'hidden';
+    }
+  });
+
 }
 
 function renderGreenPeppers() {
   // Iteration 1: set the visibility of `<section class="green-pepper">`
+  document.querySelectorAll('.green-pepper').forEach(function (onePepper) {
+    if (state.greenPeppers) {
+      onePepper.style.visibility = 'visible';
+    } else {
+      onePepper.style.visibility = 'hidden';
+    }
+  })
+
+
+
 }
 
+
+
+
+
 function renderWhiteSauce() {
   // Iteration 2: add/remove the class "sauce-white" of `<section class="sauce">`
+  document.querySelectorAll('.sauce').forEach(function (sauce) {
+    if (state.whiteSauce === false) {
+      sauce.removeAttribute("sauce-white");
+    } else {
+      sauce.setAttribute("class", "sauce-white");
+    }
+  })
+
+
 }
 
 function renderGlutenFreeCrust() {
   // Iteration 2: add/remove the class "crust-gluten-free" of `<section class="crust">`
+
+  document.querySelectorAll('.crust').forEach(function (crust) {
+    if (state.glutenFreeCrust === false) {
+      crust.removeAttribute("crust-gluten-free");
+    } else {
+      sauce.setAttribute("class", "crust-gluten-free");
+    }
+  })
+
+
+
+
 }
 
 function renderButtons() {
   // Iteration 3: add/remove the class "active" of each `<button class="btn">`
+
+  document.querySelectorAll('.btn').forEach(function (btn) {
+    btn.removeAttribute("active");
+  })
 }
```

Comment:
This is a solution example for this function:

`function renderButtons() { var ingArr = Object.keys(state); document.querySelectorAll(".btn").forEach(function(item, index) { if (state[ingArr[index]]) { item.classList.add("active"); } else { item.classList.remove("active"); } }); }`